temporian/implementation/numpy/operators/test/glue_test.py

```diff
@@ -118,6 +119,51 @@ def test_duplicate_feature(self):
             )
             _ = GlueOperator(input_0=n1, input_1=n2)
 
+    def test_order_unchanged(self):
+        """Tests that input evsets' order is kept.
+
+        Regression test for failing case where glue misordered its inputs when
+        more than 10, because of sorted() being called over a list where
+        "input_10"  was interpreted as less than "input_2".
+        """
+        evset_0 = event_set(
+            timestamps=[1],
+            features={
+                "f0": np.asarray([1], dtype=np.int64),
+            },
+        )
+        evset_1 = evset_0.rename("f1")
+        evset_2 = evset_0.rename("f2")
+        evset_3 = evset_0.rename("f3")
+        evset_4 = evset_0.rename("f4")
+        evset_5 = evset_0.rename("f5")
+        evset_6 = evset_0.rename("f6")
+        evset_7 = evset_0.rename("f7")
+        evset_8 = evset_0.rename("f8")
+        evset_9 = evset_0.rename("f9")
+
+        evset_10 = event_set(
+            timestamps=[1],
+            features={
+                "f10": np.asarray([1], dtype=np.int32),
+            },
+            same_sampling_as=evset_0,
+        )
+
+        glue(
+            evset_0,
+            evset_1,
+            evset_2,
+            evset_3,
+            evset_4,
+            evset_5,
+            evset_6,
+            evset_7,
+            evset_8,
+            evset_9,
+            evset_10,
+        )
```

Comment:
I agree that the order should be preserved from the input features, anything else seems unexpected.
In this test, I'd explicitly check that a non-alphabetical order is also preserved.